### PR TITLE
fix(python): preserve empty tool results

### LINKED ISSUE #14359: Python: Empty tool results lose correlation and fail serialization
**Describe the bug**

`ChatHistory.add_tool_message()` drops an empty-string tool result instead of preserving it as `FunctionResultContent`. The appended message still has role `tool`, but its `items` list is empty and the supplied `tool_call_id` is lost. Serializing that otherwise ordinary history message then raises `IndexError` because `ChatMessageContent.to_dict()` expects a tool message to contain a function result.

An empty string is a valid tool result for cases such as a successful command with no stdout or a no-op tool. The helper documents `content` as a string and requires `tool_call_id`; it should retain both even when the string is empty.

**To reproduce**

```python
from semantic_kernel.contents.chat_history import ChatHistory

history = ChatHistory()
history.add_tool_message("", tool_call_id="call_123")

message = history.messages[-1]
print(message.role.value, len(message.items))
print(message.to_dict())
```

On current `main` (`3438d882`):

```text
tool 0
IndexError: list index out of range
```

No model, provider, credentials, or network request is involved.

**Root cause and proposed scope**

`ChatHistory._prepare_for_add()` currently constructs `FunctionResultContent` only when `content` is truthy:

```python
if role == AuthorRole.TOOL and content and not items:
```

Changing that predicate to `content is not None` preserves the existing behavior while allowing a valid empty result to retain `result=""`, `id`, and `call_id`. A focused regression should assert both the `FunctionResultContent` fields and the serialized tool-message dictionary.

This is related to, but distinct from, #13678: that PR improves an Azure adapter's error for manually constructed tool messages with no items. This issue is about preventing the public `ChatHistory.add_tool_message()` helper from creating that invalid shape when its documented string input is empty.

**Expected behavior**

The helper should append one `FunctionResultContent` with the empty result and supplied call ID, and `message.to_dict()` should produce a valid tool message instead of raising.

**Platform**

- Language: Python
- Source: current repository `main` at `3438d882`
- AI model: not applicable
- OS: macOS; provider-free reproduction

I can submit the focused predicate change and regression once maintainers confirm this contract.

### PR DESCRIPTION
**Pull request by mikemikimike**
### Motivation and Context

Fixes #14359.

`ChatHistory.add_tool_message()` currently treats an empty string as absent content. This creates a tool message with no `FunctionResultContent`, loses the supplied tool-call correlation, and makes `ChatMessageContent.to_dict()` raise `IndexError`.

An empty string is a valid result for a successful command with no output or a no-op tool.

### Description

Change the tool-message preparation path to distinguish `None` from an empty string. Empty string results now create the same `FunctionResultContent` as other string results, preserving the result, `id`, `call_id`, and optional function name. Non-tool messages retain their existing truthiness behavior.

Added a focused regression test that verifies the content object fields and successful serialized tool-message dictionary.

This is backward compatible: non-empty tool results and all existing message forms are unchanged, and the fix only makes the documented string input work for the valid empty-string boundary case.

### Validation

- `uv run pytest tests/unit/contents/test_chat_history.py -k empty_tool_message -q --basetemp D:\\project\\tscodex\\github_pr\\tmp\\semantic-kernel-14359-pytest` — passed (1 test)
- `uv run pytest tests/unit/contents/test_chat_history.py -q --basetemp D:\\project\\tscodex\\github_pr\\tmp\\semantic-kernel-14359` — passed (56 tests)
- `uv run ruff check semantic_kernel/contents/chat_history.py tests/unit/contents/test_chat_history.py` — passed
- `uv run ruff format --check semantic_kernel/contents/chat_history.py tests/unit/contents/test_chat_history.py` — passed
- `uv run mypy semantic_kernel/contents/chat_history.py` — passed
- `uv run pre-commit run --files semantic_kernel/contents/chat_history.py tests/unit/contents/test_chat_history.py` — passed
- `git diff --check` — passed

The full Python unit suite was not run because this is a localized content-model change; no model, provider, credentials, database, or external service is required.

### Contribution Checklist

- [x] The code builds clean without any errors or warnings for the touched Python module
- [x] The PR follows the SK Contribution Guidelines and the pre-submission checks pass
- [x] The focused unit tests pass, including a regression test
- [x] The change is limited to the reported bug

### AI Disclosure

This contribution was prepared with AI assistance and reviewed against the repository source, issue reproduction, and local test results.

### CODE REVIEW

File: python/semantic_kernel/contents/chat_history.py [hunk 1]
```diff
@@ -215,7 +215,7 @@ def add_message(
 
     def _prepare_for_add(
         self, role: AuthorRole, content: str | None = None, items: list[KernelContent] | None = None, **kwargs: Any
-    ) -> dict[str, str]:
+    ) -> dict[str, Any]:
```
Comment (mikemikimike) [inline python/semantic_kernel/contents/chat_history.py:218]: Addressed in commit f53a8be2d1bf0176f1c17ab3804f33956e929c3b by changing _prepare_for_add return annotation to dict[str, Any]. This reflects the actual role and items values returned by the helper.

**Review by Ricky-7-Yan — APPROVED** [review]
Validated the reported failure and the fix independently on Python 3.12. Before the guard change, an empty tool result leaves items empty and 	o_dict() raises IndexError; with this PR's exact predicate/type-annotation change, the empty result preserves FunctionResultContent, id/call_id, and serializes correctly. I also checked the adjacent boundaries: None remains absent, list-based tool content is unchanged, non-tool empty-string behavior is unchanged, and ordinary non-empty tool results still serialize identically. The focused change looks correct.
